src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -677,12 +673,7 @@ public Visitable visit(Parameter parameter, Void p) {
       } else {
         // since it is unsolved, it could not be a primitive type
         @ClassGetSimpleName String className = parameter.getType().asClassOrInterfaceType().getName().asString();
-        if (classAndPackageMap.containsKey(className)) {
-          UnsolvedClass newClass = new UnsolvedClass(className, classAndPackageMap.get(className));
-          updateMissingClass(newClass);
-        } else {
-          throw new RuntimeException("Unexpected class: " + className);
-        }
+        updateMissingClass(createUnsolvedClass(className));
```

Comment:
The call to `updateMissingClass` should be inside the `createUnsolvedClass` method, since you call it on the result at every call site. The name of the method might need to change accordingly.